packages/flutter/test/material/dialog_test.dart

```diff
@@ -607,4 +607,43 @@ void main() {
     expect(find.text('0'), findsNothing);
     expect(find.text('1'), findsNothing);
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/28505.
+  testWidgets('showDialog only gets Theme from context on the first call', (WidgetTester tester) async {
+    Widget buildFrame(Key builderKey) {
+      return MaterialApp(
+        home: Center(
+          child: Builder(
+            key: builderKey,
+            builder: (BuildContext outerContext) {
+              return RaisedButton(
+                onPressed: () {
+                  showDialog<void>(
+                    context: outerContext,
+                    builder: (BuildContext innerContext) {
+                      return const AlertDialog(title: Text('Title'));
+                    },
+                  );
+                },
+                child: const Text('Show Dialog'),
+              );
+            },
+          ),
+        ),
+      );
+    }
+
+    // First time build.
+    await tester.pumpWidget(buildFrame(UniqueKey()));
+
+    // Open the dialog.
+    await tester.tap(find.byType(RaisedButton));
+    await tester.pumpAndSettle();
+
+    // Second time build (deactivate old context).
+    await tester.pumpWidget(buildFrame(UniqueKey()));
+
+    // Before the fix, app crashes when we try to open the dialog again.
```

Comment:
I don't think this is quite right.  The previous step will cause the button's onPressed closure to be recreated, it will refer to the new outerContext value.

The failure mode occurred when the already-visible dialog was rebuilt, because that caused it to refer to the old deactivated context.  Is it enough to just pump one frame here?

Reply:
You're totally right, it just took another `tester.pump` to get the crash.  Fixed.